plugins/woocommerce/includes/gateways/cod/class-wc-gateway-cod.php

```diff
@@ -203,10 +203,10 @@ function ( $shipping_method ) {
 	 */
 	private function is_accessing_settings() {
 		if ( is_admin() ) {
-			// phpcs:disable WordPress.Security.NonceVerification
-			if ( ! isset( $_REQUEST['page'] ) || 'wc-settings' !== $_REQUEST['page'] ) {
```

Comment:
Changing this from `$_REQUEST` to `$_GET` feels risky, as it would previously return early for `POST`ed requests as well.

Reply:
Good catch, folks! Updated the implementation in https://github.com/woocommerce/woocommerce/pull/59069/commits/ec8e38eb4295723719bde950046e92421beacfc0